### ISSUE TITLE
catalogs with @ packages not working with `pnpm update -i`

### ISSUE THREAD

**Issue by wong2** [pnpm/pnpm#8566]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

9.11.0

### Which area(s) of pnpm are affected? (leave empty if unsure)

Dependencies resolver

### Reproduction steps

I have this catalog config in my `pnpm-workspace.yaml`:

```
 catalog:
   '@headlessui/react': 2.1.2
```

### Describe the Bug

When running `pnpm update -i`, I get the following error:

`ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  @headlessui/react@catalog: isn't supported by any available resolver.`

### Expected Behavior

I can update interactively normally.

### Which Node.js version are you using?

22

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by mazei513** [comment]
Also facing this issue. Had `typescript` set up in catalog. Workspace structure was

```
|- pnpm-workspace.yaml
|- project-a/
   |- package.json
|- project-b/
   |- package.json
```

When running `pnpm up -i` in `project-a/`, got the error `ERR_PNPM_SPEC_NOT_SUPPORTED_BY_ANY_RESOLVER  typescript@catalog: isn't supported by any available resolver.`. Doing `pnpm up` works fine.

**Comment by wong2** [comment]
@mazei513 So it seems unrelated to the @ symbol.

**Comment by trydofor** [comment]
is this about https://pnpm.io/catalogs#caveats ?

> Caveats
> The pnpm update command does not yet support catalogs.
>
> To update dependencies defined in pnpm-workspace.yaml, newer version ranges will need to be  chosen manually until a future version of pnpm handles this.

**Comment by mazei513** [comment]
For me it was that `pnpm up` works as expected where it tries to update all dependencies except for those in the catalog. It is specifically `pnpm up -i` that didn't work at all. The expectation was that it would bring up the interactive picker with dependencies that are not in the catalog.

**Comment by gluxon** [comment]
Thanks for reporting @wong2 and @mazei513. I'll take a look!

**Comment by hakimio** [comment]
@gluxon any update on the issue? This prevents us from being able to use the `catalog` feature.

**Comment by lbxa** [comment]
> @gluxon any update on the issue? This prevents us from being able to use the `catalog` feature.

Agreed. This makes the `catalog` feature unusable and slightly superfluous. No one uses catalog's unless they care about standardising deps, and at that point they'd also be using the `pnpm up` command to scan for outdated deps.

**Comment by gluxon** [comment]
I'll take a look this weekend!
